src/library/setlogfeature.cpp

```diff
@@ -17,11 +17,11 @@ SetlogFeature::SetlogFeature(QObject* parent,
                              TrackCollection* pTrackCollection)
         : BasePlaylistFeature(parent, pConfig, pTrackCollection, "SETLOGHOME"),
           m_playlistId(-1) {
+    // we cleanup old history playlists that don't have any tracks
+    m_playlistDao.removeEmptyPlaylists(PlaylistDAO::HiddenType::PLHT_SET_LOG);
```

Comment:
It would be a good idea to leave a note in the code. It is not obvious that the order is crucial here, e.g. something like // Cleanup database BEFORE creating and connecting the corresponding table model